model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/authorization/policy/entity/JpaPolicyEntity.java

```diff
@@ -0,0 +1,364 @@
+        this.name = name;
+        this.owner = owner;
+        this.type = type;
+        this.metadata = null;
+    }
+
+    public boolean isMetadataInitialized() {
+        return metadata != null;
+    }
+
+    @Override
+    public Integer getEntityVersion() {
+        if (isMetadataInitialized()) return metadata.getEntityVersion();
+        return entityVersion;
+    }
+
+    @Override
+    public void setEntityVersion(Integer entityVersion) {
+        metadata.setEntityVersion(entityVersion);
+    }
+
+    @Override
+    public Integer getCurrentSchemaVersion() {
+        return CURRENT_SCHEMA_VERSION_AUTHZ;
+    }
+
+    @Override
+    public int getVersion() {
+        return version;
+    }
+
+    @Override
+    public String getId() {
+        return id == null ? null : id.toString();
+    }
+
+    @Override
+    public void setId(String id) {
+        String validatedId = UuidValidator.validateAndConvert(id);
+        this.id = UUID.fromString(validatedId);
+    }
+
+    @Override
+    public String getRealmId() {
+        if (isMetadataInitialized()) return metadata.getRealmId();
+        return realmId;
+    }
+
+    @Override
+    public void setRealmId(String realmId) {
+        metadata.setRealmId(realmId);
+    }
+
+    @Override
+    public String getResourceServerId() {
+        if (isMetadataInitialized()) return metadata.getResourceServerId();
+        return resourceServerId == null ? null : resourceServerId.toString();
+    }
+
+    @Override
+    public void setResourceServerId(String resourceServerId) {
+        metadata.setResourceServerId(resourceServerId);
+    }
+
+    @Override
+    public String getName() {
+        if (isMetadataInitialized()) return metadata.getName();
+        return name;
+    }
+
+    @Override
+    public void setName(String name) {
+        metadata.setName(name);
+    }
+
+    @Override
+    public String getType() {
+        if (isMetadataInitialized()) return metadata.getType();
+        return type;
+    }
+
+    @Override
+    public void setType(String type) {
+        metadata.setType(type);
+    }
+
+    @Override
+    public String getDescription() {
+        return metadata.getDescription();
+    }
+
+    @Override
+    public void setDescription(String description) {
+        metadata.setDescription(description);
+    }
+
+    @Override
+    public DecisionStrategy getDecisionStrategy() {
+        return metadata.getDecisionStrategy();
+    }
+
+    @Override
+    public void setDecisionStrategy(DecisionStrategy decisionStrategy) {
+        metadata.setDecisionStrategy(decisionStrategy);
+    }
+
+    @Override
+    public Logic getLogic() {
+        return metadata.getLogic();
+    }
+
+    @Override
+    public void setLogic(Logic logic) {
+        metadata.setLogic(logic);
+    }
+
+    @Override
+    public Map<String, String> getConfigs() {
+        return config.stream().collect(Collectors.toUnmodifiableMap(JpaPolicyConfigEntity::getName, JpaPolicyConfigEntity::getValue));
```

Comment:
This API would require JDK11, and for now the pom.xml of this module doesn't allow for JDK11. 

Either use a different API, or update the pom.xml to allow JDK11 (like it is done already with the Quarkus distribution).

Reply:
I've changed an API to be JDK8 compatible to be on a safe site. E.g. here https://github.com/keycloak/keycloak/blob/main/docs/building.md it is stated it requires JDK8 or higher. If we decide we'd like to have new storage modules only JDK11+ we can revisit it, but I'd say in different PR.